Library/Formula/squid-deb-proxy.rb

```diff
@@ -0,0 +1,174 @@
+class SquidDebProxy < Formula
+  desc "APT repo caching proxy based on Squid."
+  homepage "https://launchpad.net/squid-deb-proxy"
+  url "https://mirrors.ocf.berkeley.edu/debian/pool/main/s/squid-deb-proxy/squid-deb-proxy_0.8.14.tar.gz"
+  sha256 "d30aa9d8185c1e424608deb307f8e92aee8270acccc3c0b26af1d03a2feae92b"
+
+  depends_on "squid"
+
+  # assorted changes that should be submitted upstream
+  patch :DATA
+
+  def install
+    # fix up issues specific to Homebrew
+    inreplace "squid-deb-proxy.conf" do |s|
+      # fix paths with the HOMEBREW_PREFIX
+      s.gsub! %r{([\s='"])/etc/}, "\\1#{etc}/"
+      s.gsub! %r{([\s='"])/var/}, "\\1#{var}/"
+    end
+
+    inreplace "init-common.sh" do |s|
+      # install should be as the calling user
+      s.gsub! /(\binstall\b.*?)(\s+-[og]\s*\S+)+/, "\\1"
+
+      # fix paths with the HOMEBREW_PREFIX
+      s.gsub! "/usr/sbin/squid", "#{HOMEBREW_PREFIX}/opt/squid/sbin/squid"
+      s.gsub! %r{([\s='"])/etc/}, "\\1#{etc}/"
+      s.gsub! %r{([\s='"])/var/}, "\\1#{var}/"
+    end
+
+    # pre-installation from debian/rules
+    # we take both sets because we're not either distro
+    IO.write(
+      "mirror-dstdomain.acl",
+      IO.read("mirror-dstdomain.acl.Debian") +
+        IO.read("mirror-dstdomain.acl.Ubuntu")
+    )
+
+    system "make", "install", "DESTDIR=#{prefix}"
+
+    # additional installation from debian/squid-deb-proxy.install
+    mkdir_p "#{prefix}/usr/share/squid-deb-proxy"
+    cp "init-common.sh", "#{prefix}/usr/share/squid-deb-proxy/"
+
+    # produce execution wrapper script
+    Pathname.new("#{sbin}/squid-deb-proxy").write(
+      <<-EOS.undent
+      #!/bin/bash
+
+      . '#{opt_prefix}/usr/share/squid-deb-proxy/init-common.sh'
+      pre_start
+
+      exec "$SQUID" -f '#{etc}/squid-deb-proxy/squid-deb-proxy.conf' "$@"
+      EOS
+    )
+
+    # produce placeholder files from debian/squid-deb-proxy.postinst
+    # these also serve to prevent the directories from being removed
+    user_conf      = "#{etc}/squid-deb-proxy/squid.conf.d/10-default.conf"
+    user_networks  = "#{etc}/squid-deb-proxy/allowed-networks-src.acl.d/10-default"
+    user_dests     = "#{etc}/squid-deb-proxy/mirror-dstdomain.acl.d/10-default"
+    user_blacklist = "#{etc}/squid-deb-proxy/pkg-blacklist.d/10-default"
+    Pathname.new(user_conf).write(
+      <<-EOS.undent
+      # #{user_conf}
+      #
+      # additional Squid configuration
+      EOS
+    )
+    Pathname.new(user_networks).write(
```

Comment:
Same deal on the `Pathname`. Should work with the same style of syntax example above.